### PR TITLE
[CVE-2022-1471][RHPAM-4700] snakeyaml from 1.32 to 2.0

### PR DESCRIPTION
**Pull request by Ginxo**
**JIRA**: 
https://issues.redhat.com/browse/RHPAM-4700

Related PRs:
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2285
- https://github.com/kiegroup/optaplanner/pull/2822

You can check Kiegroup organization repositories CI status from [Chain Status webpage](https://kiegroup.github.io/droolsjbpm-build-bootstrap/)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used locally on command line or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it. 

A general local execution could be the following one, where the tool clones all dependent projects starting from the `-sp` one and it locally applies the pull request (if it exists) in order to reproduce a complete build scenario for the provided *Pull Request*.

> **Note:** the tool considers multiple *Pull Requests* related to each other if their branches (generally in the forked repositories) have the same name.

``` shell
$ build-chain-action -df 'https://raw.githubusercontent.com/${GROUP:kiegroup}/droolsjbpm-build-bootstrap/${BRANCH:main}/.ci/pull-request-config.yaml' build pr -url <pull-request-url> -sp kiegroup/kie-wb-distributions [--skipExecution]
```

> Consider changing `kiegroup/kie-wb-distributions` with the correct starting project.

</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest</b> (using <i>this</i> e.g. <b>Jenkins retest this</b> optional but no longer required)
 
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
    
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

* <b>for windows-specific os job</b> add the label `windows_check`
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by Ginxo** [comment]
FDB without https://github.com/kiegroup/optaplanner/pull/2821 -> https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/main/job/fdb/job/droolsjbpm-build-bootstrap-main.fdb/647/

**Comment by Ginxo** [comment]
Jenkins run fdb

**Comment by Ginxo** [comment]
2 test failures related to the change:

- org.kie.server.integrationtests.dmn.modelspecific.ModelSpecificIntegrationTest.test_getOAS
- org.kie.server.integrationtests.dmn.modelspecific.ModelSpecificIntegrationTest.test_getOASYAML

```
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] Caused by: java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.io.ContentReference com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(java.lang.Object)'
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:406)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at com.fasterxml.jackson.core.jackson-databind@2.12.1//com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3023)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//io.smallrye.openapi.runtime.io.OpenApiParser.parse(OpenApiParser.java:76)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//io.smallrye.openapi.runtime.OpenApiProcessor.modelFromStaticFile(OpenApiProcessor.java:101)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//org.kie.server.services.dmn.modelspecific.OASGenerator.loadOASTemplate(OASGenerator.java:178)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//org.kie.server.services.dmn.modelspecific.OASGenerator.generateOAS(OASGenerator.java:64)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//org.kie.server.services.dmn.ModelEvaluatorServiceBase.getOAS(ModelEvaluatorServiceBase.java:337)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//org.kie.server.remote.rest.dmn.ModelEvaluatorResource.getOASType(ModelEvaluatorResource.java:229)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:138)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:546)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:435)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$0(ResourceMethodInvoker.java:396)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:358)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:398)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:365)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:338)
[2023-05-10T12:12:43.415Z] [INFO] [talledLocalContainer] 	at org.jboss.resteasy.resteasy-jaxrs@3.15.1.Final//org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:440)
```

**Comment by Ginxo** [comment]
> but don't we need to upgrade this https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/pom.xml#LL77C1-L78C94 anyway ? And i guess jackson should be managed in tests [droolsjbpm-integration](https://github.com/kiegroup/droolsjbpm-integration/tree/55a57cda052f2c860326c434562655d3ab4d3be6)/[kie-server-parent](https://github.com/kiegroup/droolsjbpm-integration/tree/55a57cda052f2c860326c434562655d3ab4d3be6/kie-server-parent)/[kie-server-tests](https://github.com/kiegroup/droolsjbpm-integration/tree/55a57cda052f2c860326c434562655d3ab4d3be6/kie-server-parent/kie-server-tests)/pom.xml

https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/pom.xml#LL77C1-L78C94 is already part of this PR
what do you mean by `And i guess jackson should be managed in tests droolsjbpm-integration/kie-server-parent/kie-server-tests/pom.xml` sorry but I don't get what you exactly mean :thinking: 

Thanks @mareknovotny

**Comment by mareknovotny** [comment]
> > but don't we need to upgrade this https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/pom.xml#LL77C1-L78C94 anyway ? And i guess jackson should be managed in tests [droolsjbpm-integration](https://github.com/kiegroup/droolsjbpm-integration/tree/55a57cda052f2c860326c434562655d3ab4d3be6)/[kie-server-parent](https://github.com/kiegroup/droolsjbpm-integration/tree/55a57cda052f2c860326c434562655d3ab4d3be6/kie-server-parent)/[kie-server-tests](https://github.com/kiegroup/droolsjbpm-integration/tree/55a57cda052f2c860326c434562655d3ab4d3be6/kie-server-parent/kie-server-tests)/pom.xml
> 
> https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/pom.xml#LL77C1-L78C94 is already part of this PR 

yes, i meant that ;)

>what do you mean by `And i guess jackson should be managed in tests droolsjbpm-integration/kie-server-parent/kie-server-tests/pom.xml` sorry but I don't get what you exactly mean thinking
> 
> Thanks @mareknovotny

and yes, kie-parent is managing the jackson's version/depmgmt, but integration tests are not directly declaring jackson's libs

**Comment by Ginxo** [comment]
@tkobayas (or better contributor of your choice @tarilabs as the ModelSpecificIntegrationTest creator maybe?) 
Would it be possible to adapt?
* org.kie.server.integrationtests.dmn.modelspecific.ModelSpecificIntegrationTest.test_getOAS
* org.kie.server.integrationtests.dmn.modelspecific.ModelSpecificIntegrationTest.test_getOASYAML

thanks

**Comment by tkobayas** [comment]
Hi @tarilabs , could you kindly have a look? Maybe OpenAPI version related?

**Comment by tarilabs** [comment]
> @tkobayas (or better contributor of your choice @tarilabs as the ModelSpecificIntegrationTest creator maybe?) Would it be possible to adapt?
> 
> * org.kie.server.integrationtests.dmn.modelspecific.ModelSpecificIntegrationTest.test_getOAS
> * org.kie.server.integrationtests.dmn.modelspecific.ModelSpecificIntegrationTest.test_getOASYAML
> 
> thanks

i'm sorry i didn't really get the question @Ginxo ? iirc we use snakeyaml as inherited from jackson (for jackson yaml). what am i missing?

**Comment by Ginxo** [comment]
thanks for the support @tarilabs 
I mean `com.fasterxml.jackson.core.jackson-databind@2.12.1` is being used by wildfly during the tests, no matter the version we specify there, those `2.12.1` libraries are not even in the dependency:tree information since they are taken from target/wildfly libraries folder and additionally to it there's no wildfly version providing the expected libraries.
I guess we should workaround the problem by adapting (even skipping the tests... :thinking: )

**Comment by tarilabs** [comment]
to me it seems you are redefining a snakeyaml version which is not compatible with jackson, as i see:

```
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] Caused by: java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.io.ContentReference com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(java.lang.Object)'
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:406)
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] 	at com.fasterxml.jackson.core.jackson-databind@2.12.1//com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3023)
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//io.smallrye.openapi.runtime.io.OpenApiParser.parse(OpenApiParser.java:76)
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//io.smallrye.openapi.runtime.OpenApiProcessor.modelFromStaticFile(OpenApiProcessor.java:101)
[2023-05-10T12:32:29.174Z] [INFO] [talledLocalContainer] 	at deployment.kie-server-services.war//org.kie.server.services.dmn.modelspecific.OASGenerator.loadOASTemplate(OASGenerator.java:178)
```

instead of bumping snake yaml to v2 @Ginxo is there a 1.x version which jackson is fine with ?

**Comment by tarilabs** [comment]
> there is no snakeyaml 1.x we can use, 

Mind to explain better please? I'm also available via call if preferred.

The test uses Jackson Yaml, as it test standard way of kie server the functionality.

Disabling the test means regression.

I cannot "readapt" the test, what is failing is jackson yaml for the underlying capability of kie server service

File: pom.xml [hunk 1]
```diff
@@ -74,8 +74,8 @@
     </version.com.ahome-it.lienzo.tooling.nativetools>
     <version.com.ahome-it.lienzo.tooling.common>1.0.190-RELEASE</version.com.ahome-it.lienzo.tooling.common>
     <version.com.allen-sauer.gwt.dnd>3.3.3</version.com.allen-sauer.gwt.dnd>
-    <version.com.fasterxml.jackson>2.12.7</version.com.fasterxml.jackson>
-    <version.com.fasterxml.jackson.databind>2.12.7.1</version.com.fasterxml.jackson.databind>
+    <version.com.fasterxml.jackson>2.15.0</version.com.fasterxml.jackson>
+    <version.com.fasterxml.jackson.databind>${version.com.fasterxml.jackson}</version.com.fasterxml.jackson.databind>
```
Comment (tarilabs) [inline pom.xml:78]: i would like to admit i haven't grokked the need for this change :/

**Comment by mareknovotny** [comment]
> > there is no snakeyaml 1.x we can use,
> 
> Mind to explain better please? I'm also available via call if preferred.
> 
> The test uses Jackson Yaml, as it test standard way of kie server the functionality.
> 
> Disabling the test means regression.
> 
> I cannot "readapt" the test, what is failing is jackson yaml for the underlying capability of kie server service

if you read what https://issues.redhat.com/browse/RHPAM-4700 is about then you can get the why we need to upgrade to 2.0

if tests can't use that version we need to mark them as ignored and create jira to fix them and that will be assigned to someone who know the test use case.

**Comment by tarilabs** [comment]
(giving a spin at testing only update snakeyaml locally)

**Comment by mareknovotny** [comment]
i think it was the original automatic PR change from dependabot, which failed due other test failures in more tests. see https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2243

**Comment by tarilabs** [comment]
kie-dmn-openapi -> smallrye openapi (untouched) -> jackson yaml (untouched) -> snakeyaml (bumped) worked locally for the engine capability.

```
[INFO] org.kie:kie-dmn-openapi:jar:7.74.0-SNAPSHOT
[INFO] +- io.smallrye:smallrye-open-api-core:jar:2.1.4:compile
[INFO] |  +- org.eclipse.microprofile.openapi:microprofile-openapi-api:jar:2.0.1-RC1:compile
[INFO] |  +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-core:jar:2.12.7:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.12.7.1:compile
[INFO] |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.12.7:compile
[INFO] |  |  \- org.yaml:snakeyaml:jar:2.0:compile
```

now going to check integrating on kie-server with testing PR for CI (and locally) with: https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2290

**Comment by tarilabs** [comment]
> i think it was the original automatic PR change from dependabot, which failed due other test failures in more tests. see #2243

checking if it's a "short blanket" issue; at least for dmn engine OAS capability seems fine just the bump of snakeyaml per https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2285#issuecomment-1551321186

i would assume CI would fail as that would not please optaplanner but i would check locally if kie-server test passing with the mod (without involving OP build locally) so to investigate if a "short blanket" case.

**Comment by Ginxo** [comment]
so @tarilabs will open new set of PRs bumping snakeyaml on kie-parent and whatever the jackson version needed by optaplanner.

**Comment by tarilabs** [comment]
> so @tarilabs will open new set of PRs bumping snakeyaml on kie-parent and whatever the jackson version needed by optaplanner.

indeed one already raised, trying also locally, and looking now at the OP quickstart

**Comment by tarilabs** [comment]
i got the OP quickstart to pass test by specifying the version of snakeyaml for the quickstart specificially, that is because the quickstart does not inherit from the build bom hence it need to be specified, seems to me.

**Comment by tarilabs** [comment]
i've sent an email recap about my investigation and i will continue in the direction.

**Comment by mareknovotny** [comment]
closing this PR as there were developed different PR set and merged https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2296

**Pull request closed without merging**